File: python/1045.py
import math

def quicksort(arr):
    if(len(arr) <= 1):
        return arr
    else:
        pivot = arr[len(arr) // 2]
        left = [x for x in arr if x > pivot]
        right = [x for x in arr if x < pivot]
        middle = [x for x in arr if x == pivot]
        return quicksort(left) + middle + quicksort(right)

lados = list(map(float, input().split()))
lados_decrescente = quicksort(lados)
K = False
if(lados_decrescente[0] >= lados_decrescente[1] + lados_decrescente[2]):
    print("NAO FORMA TRIANGULO")
else:
    if(pow(lados_decrescente[0],2) == pow(lados_decrescente[1],2) + pow(lados_decrescente[2],2)):
        print("TRIANGULO RETANGULO")
    if(pow(lados_decrescente[0],2) > pow(lados_decrescente[1],2) + pow(lados_decrescente[2],2)):
        print("TRIANGULO OBTUSANGULO")
    if(pow(lados_decrescente[0],2) < pow(lados_decrescente[1],2) + pow(lados_decrescente[2],2)):
        print("TRIANGULO ACUTANGULO")
    if(lados_decrescente[0] == lados_decrescente[1] == lados_decrescente[2]):
        print("TRIANGULO EQUILATERO")
        K = True
    for i in range(2):
        if(K):
            break
        for j in range(i+1,3,1):
            if(lados_decrescente[i] == lados_decrescente[j]):
                print("TRIANGULO ISOSCELES")
                K = True
                break

File: C/1045.c
#include <stdio.h>
#include <math.h>

int main() {
    float a, b, c;


    scanf("%f %f %f", &a, &b, &c);


    if (b > a) {
        float temp = a; a = b; b = temp;
    }
    if (c > a) {
        float temp = a; a = c; c = temp;
    }
    if (c > b) {
        float temp = b; b = c; c = temp;
    }


    if (a >= b + c || a <= 0 || b <= 0 || c <= 0) {
        printf("NAO FORMA TRIANGULO\n");
        return 0;
    }

    if (pow(a, 2) == pow(b, 2) + pow(c, 2)) {
        printf("TRIANGULO RETANGULO\n");
    } else if (pow(a, 2) > pow(b, 2) + pow(c, 2)) {
        printf("TRIANGULO OBTUSANGULO\n");
    } else if (pow(a, 2) < pow(b, 2) + pow(c, 2)) {
        printf("TRIANGULO ACUTANGULO\n");
    }

    if (a == b && b == c) {
        printf("TRIANGULO EQUILATERO\n");
    } else if (a == b || a == c || b == c) {
        printf("TRIANGULO ISOSCELES\n");
    }

    return 0;
}
